lib/enumerables.rb

```diff
@@ -0,0 +1,177 @@
+module Enumerable
+  # my_each
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each>"
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i], count
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count], count
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each_with_index>"
+    end
+  end
+
+  # my_select
+  def my_select(&block)
+    return "\#<Enumerator: #{self}:my_select>" unless block
+
+    temp = []
+    to_a.my_each { |num| temp.push(num) if yield(num) }
+    temp
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return false if n.nil? || !n }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false unless x.include?(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return false unless x.is_a?(paramet) }
+    else
+      arr.my_each { |x| return false if x != paramet }
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return true if ans == true
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return true if !!n == true && !n.nil? }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return true if x.match(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return true if x.include?(paramet) }
```

Comment:
Again, your `#my_each` method doesn't seem able to handle this type of variable (`Class` variables):

```
  6) enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
     Failure/Error: arr.my_each { |x| return true if x.include?(paramet) }
     
     NoMethodError:
       undefined method `include?' for 3:Integer
     # ./lib/enumerables.rb:94:in `block in my_any?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:94:in `my_any?'
     # ./spec/enumerables_spec.rb:242:in `block (4 levels) in <top (required)>'

  7) enumerables #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
     Failure/Error: arr.my_each { |x| return true if x.include?(paramet) }
     
     TypeError:
       no implicit conversion of Class into String
     # ./lib/enumerables.rb:94:in `include?'
     # ./lib/enumerables.rb:94:in `block in my_any?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:94:in `my_any?'
     # ./spec/enumerables_spec.rb:246:in `block (4 levels) in <top (required)>'
```

Either prepare `#my_each` to handle classes, or write a new method to handle your `my_any` checks.